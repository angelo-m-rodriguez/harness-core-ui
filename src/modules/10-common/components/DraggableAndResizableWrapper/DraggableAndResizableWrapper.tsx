import React from 'react'
import { Rnd, RndResizeCallback, Props as ReactRndProps } from 'react-rnd'
import { Container } from '@wings-software/uicore'

interface DraggableAndResizableWrapperProps {
  className?: string
  onResize?: RndResizeCallback
  rndOverrideProps?: ReactRndProps
  children?: React.ReactNode
}

export const DraggableAndResizableWrapper = (
  props: DraggableAndResizableWrapperProps,
  draggableAndResizeableContainerRef: React.ForwardedRef<null | HTMLDivElement>
) => {
  const { className, onResize, rndOverrideProps } = props

  return (
    <Rnd
      bounds="window"
      enableResizing={{
        top: true,
        bottom: true,
        left: true,
        right: true,
        topRight: false,
        bottomRight: false,
        bottomLeft: false,
        topLeft: false
      }}
      {...rndOverrideProps}
      className={className}
      onResize={onResize}
    >
      <Container style={{ height: '100%' }} ref={draggableAndResizeableContainerRef}>
        {props.children}
      </Container>
    </Rnd>
  )
}

export const DraggableAndResizableWrapperWithRef = React.forwardRef(DraggableAndResizableWrapper)
