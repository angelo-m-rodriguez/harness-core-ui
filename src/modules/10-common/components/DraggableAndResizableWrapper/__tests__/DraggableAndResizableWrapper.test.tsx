import React from 'react'
import { render } from '@testing-library/react'
import { DraggableAndResizableWrapperWithRef } from '../DraggableAndResizableWrapper'

const RND_OVERRIDE_PROPS = {
  default: {
    x: 50,
    y: 30,
    width: 480,
    height: 500
  }
}

describe('Draggable and Resizable test', () => {
  test('snapshot testing 1', () => {
    const ref = React.createRef() as React.ForwardedRef<null | HTMLDivElement>
    const { container } = render(
      <DraggableAndResizableWrapperWithRef rndOverrideProps={RND_OVERRIDE_PROPS} ref={ref}>
        <div>Draggable and resizable div</div>
      </DraggableAndResizableWrapperWithRef>
    )
    expect(container).toMatchSnapshot()
  })

  test('snapshot testing with additional props', () => {
    const ref = React.createRef() as React.ForwardedRef<null | HTMLDivElement>
    const { container } = render(
      <DraggableAndResizableWrapperWithRef
        className="draggableAndResizableWrapper"
        rndOverrideProps={{ ...RND_OVERRIDE_PROPS, minWidth: 250, minHeight: 250, bounds: '#parent-id' }}
        ref={ref}
      >
        <div>Draggable and resizable div</div>
      </DraggableAndResizableWrapperWithRef>
    )
    expect(container).toMatchSnapshot()
  })
})
