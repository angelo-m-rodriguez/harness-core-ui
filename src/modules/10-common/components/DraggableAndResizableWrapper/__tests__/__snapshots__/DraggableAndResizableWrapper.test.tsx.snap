// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draggable and Resizable test snapshot testing 1 1`] = `
<div>
  <div
    class="react-draggable"
    style="position: absolute; user-select: auto; width: 480px; height: 500px; display: inline-block; top: 0px; left: 0px; cursor: move; transform: translate(100px,60px); max-width: 9007199254740991px; max-height: 9007199254740991px; box-sizing: border-box; flex-shrink: 0;"
  >
    <div
      class="StyledProps--font StyledProps--main"
      style="height: 100%;"
    >
      <div>
        Draggable and resizable div
      </div>
    </div>
    <div>
      <div
        class=""
        style="position: absolute; user-select: none; width: 100%; height: 10px; top: -5px; left: 0px; cursor: row-resize;"
      />
      <div
        class=""
        style="position: absolute; user-select: none; width: 100%; height: 10px; bottom: -5px; left: 0px; cursor: row-resize;"
      />
      <div
        class=""
        style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; left: -5px; cursor: col-resize;"
      />
      <div
        class=""
        style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; right: -5px; cursor: col-resize;"
      />
    </div>
  </div>
</div>
`;

exports[`Draggable and Resizable test snapshot testing with additional props 1`] = `
<div>
  <div
    class="draggableAndResizableWrapper react-draggable"
    style="position: absolute; user-select: auto; width: 480px; height: 500px; display: inline-block; top: 0px; left: 0px; cursor: move; transform: translate(100px,60px); max-width: 9007199254740991px; max-height: 9007199254740991px; min-width: 250px; min-height: 250px; box-sizing: border-box; flex-shrink: 0;"
  >
    <div
      class="StyledProps--font StyledProps--main"
      style="height: 100%;"
    >
      <div>
        Draggable and resizable div
      </div>
    </div>
    <div>
      <div
        class=""
        style="position: absolute; user-select: none; width: 100%; height: 10px; top: -5px; left: 0px; cursor: row-resize;"
      />
      <div
        class=""
        style="position: absolute; user-select: none; width: 100%; height: 10px; bottom: -5px; left: 0px; cursor: row-resize;"
      />
      <div
        class=""
        style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; left: -5px; cursor: col-resize;"
      />
      <div
        class=""
        style="position: absolute; user-select: none; width: 10px; height: 100%; top: 0px; right: -5px; cursor: col-resize;"
      />
    </div>
  </div>
</div>
`;
